clippy.toml

```diff
@@ -0,0 +1 @@
+msrv = "1.36.0"
```

Comment:
Do we need some remark that this line has to be updated along other files? Apart from that, I'm fine with this.

Reply:
It depends. I can't find any word about the msrv, apart from the github workflow, so first step would be to mention it in the readme I guess?
Second: there's the [`rust-version`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field) field in the `Cargo.toml` file that can be used as well. A clippy PR is pending to parse that field in addition to the `clippy.toml` file: https://github.com/rust-lang/rust-clippy/pull/8774

As soon as that lands, the `clippy.toml` can be deleted again, so I don't think that that should be documented? 😅